moit-api/src/main/kotlin/com/mashup/moit/fine/facade/FineFacade.kt

```diff
@@ -1,7 +1,40 @@
 package com.mashup.moit.fine.facade
 
+import com.mashup.moit.domain.fine.FineService
+import com.mashup.moit.domain.study.StudyService
+import com.mashup.moit.domain.user.UserService
+import com.mashup.moit.fine.controller.dto.FineListResponse
+import com.mashup.moit.fine.controller.dto.FineResponseForListView
 import org.springframework.stereotype.Component
 
 @Component
-class FineFacade {
+class FineFacade(
+    private val fineService: FineService,
+    private val studyService: StudyService,
+    private val userService: UserService
+) {
+    fun getFineList(moitId: Long): FineListResponse {
+        val fineList = fineService.getFineListByMoitId(moitId)
+        if (fineList.isEmpty()) {
+            return FineListResponse.emptyFineList()
+        }
+
+        val fineUserIds = fineList.map { it.userId }.distinct()
+        val fineUsersMap = userService.findUsersByIds(fineUserIds).associateBy { it.id }
+
+        val fineStudyIds = fineList.map { it.studyId }.distinct()
+        val fineStudyMap = studyService.findByStudyIds(fineStudyIds).associateBy { it.id }
+
+        val totalFineAmount = fineList.sumOf { it.amount }
+        val (fineComplete, fineNotYet) = fineList.partition { it.isApproved }
+            .let { (approvedFineList, isNotApprovedFineList) ->
+                approvedFineList.map {
+                    FineResponseForListView.of(it, fineUsersMap[it.userId]!!, fineStudyMap[it.studyId]!!)
```

Comment:
```suggestion
                    FineResponseForListView.of(it, fineUsersMap.getValue(it.userId), fineStudyMap.getValue(it.studyId))
```
취향이지만 `getValue()` 써서 `!!` 를 사용하지 않는 방법도 생각해 볼 수 있겠습니다..!

Reply:
맞아요 성윤씨가 올린 PR 보고 바꿔야지라고 생각했는데, 코멘트가 먼저 달렸군요 감사합니다!